### ISSUE TITLE
Deployment via windows breaks folder paths on S3

### ISSUE THREAD

**Issue by cloudandcode** [fernando-mc/serverless-finch#51]
This is a:

* Bug report

---

### Expected behavior

When using Windows to deploy a dist folder that contains sub-directories, the file paths should translate to folders in S3.

ie. a file in dist\js\app.js within Windows should result in app.js being placed within a 'js' folder in the S3 bucket so that the relative path to that file becomes /js/app.js.

### Actual behavior

Using the example above, a file in dist\js\app.js is written as js\app.js within the S3 bucket and the links to that file appears as /js%5Capp.js (instead of /js/app.js) and are therefore unreachable via the expected url, resulting in 404 errors.

### Steps to reproduce

Deploy a distribution directory that contains sub-directories from a Windows machine.

* **Operating system:** Windows 10 
* **serverless-finch version:** 2.1.0

### Proposed fix

Replacing the Windows file separator '\' with the Posix separator '/' on line 86 in /lib/upload.js does the trick for me.

`fileKey: filePath.replace(clientRoot, '').replace(/\\/g, '/'),`

**Comment by fernando-mc** [comment]
@cloudandcode Makes sense. Can you open the PR? I can can merge & cut a patch tomorrow.

**Comment by linusmarco** [comment]
@cloudandcode maybe we can make this fix a bit more general? Something like:
```
fileKey: path.normalize(filePath.replace(clientRoot, '')).split(path.sep).join('/'),
```

That way it's not a windows-only hack (that could screw things up for POSIX systems), but rather a generalized path -> AWS FileKey conversion.

**Comment by fernando-mc** [comment]
@linusmarco Do you have an example of when that fix might break things? I thought it would only really change anything when a backslash was used?

From how I understand the suggestion it would just turn all of the`\` characters into `/` characters right?

If there's a more comprehensive way to do it with path separators and joining with the '/' character let's go for it.

Either way, I'll try to review and merge the evening I see a PR that fixes this and doesn't break my manual tests.

**Comment by tahir-mm** [comment]
@fernando-mc  can we have this fix please, we are also facing same issue. 

Scenario is, We have static site with folder structure, when we deploy to AWS bucket from window, its not creating folder inside AWS bucket but creating files at the root of AWS bucket with '/' containing name.

**Comment by fernando-mc** [comment]
@tahir-mm Thanks for the heads up! This appears to be a breaking change from the significant refactor we did for version 2. Feel free to use v1 while we implement fixes for v2. 

@linusmarco I think the proposed fix should work. It appears similar to what we used to do. You can see it in action here: https://github.com/fernando-mc/serverless-finch/blob/74774bec67944658514f4850dada0b1148a3e244/index.js#L306

@cloudandcode @linusmarco @tahir-mm I'm a bit busy this week so any of you please feel free to open a PR and comment back here. I can review and cut a new version but I need a PR for the fix.

**Comment by linusmarco** [comment]
@fernando-mc And yes, we do use the backslash replace elsewhere, but it's wrong there as well. I'll also fix that in the PR. 

We've probably just not run into issues with this because it would be rare for a UNIX path to actually contain a backslash, but it theoretically could and we should be able to handle it.

**Comment by fernando-mc** [comment]
@linusmarco ahh makes sense. I guess I hadn’t considered that (weird, but possible) case. Your proposed fix makes sense.

**Comment by cloudandcode** [comment]
@fernando-mc Sorry, ended up migrating to Firebase hosting to serve my spa, much easier / quicker / feature packed than S3 / CloudFront.

**Comment by fernando-mc** [comment]
@cloudandcode and@tahir-mm  this is now addressed in `v2.1.1` just pushed the new version.

Thanks to @linusmarco!
